### ISSUE TITLE
Preprocessing error if start_time is an np.int64

### ISSUE THREAD

**Issue by louisfh** [kitzeslab/opensoundscape#778]
Weird bug being caused by `datetime.timedelta(seconds=s)` not handling `np.int64` nicely

This is a potentially annoying bug for users, because if users call `BoxedAnnotations.one_hot_clip_labels(clip_duration=x)`, where x is an integer, they will then run into this error when  trying to preprocess samples from the dataframe produced by this.

Here's an example traceback:

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
File ~/opt/anaconda3/envs/opso-0.9.1/lib/python3.9/site-packages/opensoundscape/preprocess/preprocessors.py:147, in BasePreprocessor.forward(self, sample, break_on_type, break_on_key, bypass_augmentations, trace)
    146 # perform the action (modifies the AudioSample in-place)
--> 147 action.go(sample)
    149 if trace:  # user requested record of preprocessing steps
    150     # save the current state of the sample's data
    151     # (trace is a Series with index matching self.pipeline)

File ~/opt/anaconda3/envs/opso-0.9.1/lib/python3.9/site-packages/opensoundscape/preprocess/actions.py:151, in AudioClipLoader.go(self, sample, **kwargs)
    150 duration = None if sample.duration is None else sample.duration
--> 151 sample.data = self.action_fn(
    152     sample.data, offset=offset, duration=duration, **dict(self.params, **kwargs)
    153 )

File ~/opt/anaconda3/envs/opso-0.9.1/lib/python3.9/site-packages/opensoundscape/audio.py:356, in Audio.from_file(cls, path, sample_rate, resample_type, dtype, load_metadata, offset, duration, start_timestamp, out_of_bounds_mode)
    355     if "recording_start_time" in metadata and offset > 0:
--> 356         metadata["recording_start_time"] += datetime.timedelta(seconds=offset)
    358 return cls(samples, sr, resample_type=resample_type, metadata=metadata)

TypeError: unsupported type for timedelta seconds component: numpy.int64

The above exception was the direct cause of the following exception:

PreprocessingError                        Traceback (most recent call last)
Cell In[47], line 10
      8 dataset.bypass_augmentations = True
      9 #3
---> 10 tensors = [dataset[i].data for i in range(9)]
     11 sample_labels = [list(dataset[i].labels[dataset[i].labels==1].index) for i in range(9)]
     12 _ = show_tensor_grid(tensors, 3, labels=sample_labels)

Cell In[47], line 10, in <listcomp>(.0)
      8 dataset.bypass_augmentations = True
      9 #3
---> 10 tensors = [dataset[i].data for i in range(9)]
     11 sample_labels = [list(dataset[i].labels[dataset[i].labels==1].index) for i in range(9)]
     12 _ = show_tensor_grid(tensors, 3, labels=sample_labels)

File ~/opt/anaconda3/envs/opso-0.9.1/lib/python3.9/site-packages/opensoundscape/ml/datasets.py:103, in AudioFileDataset.__getitem__(self, idx, break_on_key, break_on_type)
    100 sample = AudioSample.from_series(self.label_df.iloc[idx])
    102 # preprocessor.forward will raise PreprocessingError if something fails
--> 103 sample = self.preprocessor.forward(
    104     sample,
    105     bypass_augmentations=self.bypass_augmentations,
    106     break_on_key=break_on_key,
    107     break_on_type=break_on_type,
    108 )
    110 return sample

File ~/opt/anaconda3/envs/opso-0.9.1/lib/python3.9/site-packages/opensoundscape/preprocess/preprocessors.py:164, in BasePreprocessor.forward(self, sample, break_on_type, break_on_key, bypass_augmentations, trace)
    160                 sample.trace[k] = sample.data
    162 except Exception as exc:
    163     # treat any exceptions raised during forward as PreprocessingErrors
--> 164     raise PreprocessingError(
    165         f"failed to preprocess sample from path: {sample.source}"
    166     ) from exc
    168 # remove temporary attributes from sample
    169 del sample.preprocessor, sample.target_duration

PreprocessingError: failed to preprocess sample from path: ./nobo_annotations/noboAudio/msdw0063_20220529_100000.WAV
```

**Comment by louisfh** [comment]
This seems to be specific to particular versions of datetime (and/or possibly Numpy?).
I can't recreate this issue with:
```
Python 3.9.16  #datetime is a builtin module
Numpy 1.24.4 
```

**Comment by louisfh** [comment]
Also works fine with:

```
Python  3.8.18
Numpy 1.24.4
```

This stackoverflow answer https://stackoverflow.com/a/66700638/6130437 suggests it was a change made in Python that now allows datetime to handle `np.ints`. Which makes me think Pat (who encountered this issue) had a version of Python <3.8.18. I have not been able to identify which Python version release fixed this issue. @sammlapp I don't think it's worth trying to find out which Python version causes this.

**Comment by louisfh** [comment]
I suggest we close the issue, or change our dependencies to:
```
Python >= 3.8.18
```

**Comment by sammlapp** [comment]
Hm, maybe we'll just move to python >=3.9 in the 0.10.0 release since tensorflow and crowsetta are also complaining about python 3.8 (we are one year from 3.8's "end of life")

**Comment by sammlapp** [comment]
We believe this issue is resolved by requiring python >=3.9. The issue can be re-opened if it is encountered with python >=3.9
